src/lib/distances.jl

```diff
@@ -79,11 +79,16 @@ end
 end
 
 @adjoint function pairwise(::Euclidean, X::AbstractMatrix; dims=2)
-  D, back = pullback(X -> pairwise(SqEuclidean(), X; dims = dims), X)
-  D .= sqrt.(D)
-  return D, function(Δ)
-    Δ = Δ ./ (2 .* max.(D, eps(eltype(D))))
-    Δ[diagind(Δ)] .= 0
```

Comment:
I wonder whether this worked in higher order AD

Reply:
I can't imagine that it did, because mutation involved in `Δ[diagind(Δ)] .= 0`